### PR TITLE
VTCCA-6132 implement govuk file upload component 

### PR DESCRIPTION
**Pull request by zachhall1234**

### CODE REVIEW

File: app/controllers/propertyLinking/DeclarationController.scala [hunk 1]
```diff
@@ -72,18 +72,17 @@ class DeclarationController @Inject()(
...
+          case (RatesBillType, Owner | OwnerOccupier | Occupier)        => EvidenceChoices.RATES_BILL
+          case (ServiceCharge, Owner | OwnerOccupier | Occupier)        => EvidenceChoices.SERVICE_CHARGE
+          case (StampDutyLandTaxForm, Owner | OwnerOccupier | Occupier) => EvidenceChoices.STAMP_DUTY
+          case (LandRegistryTitle, Owner | OwnerOccupier | Occupier)    => EvidenceChoices.LAND_REGISTRY
+          case (WaterRateDemand, Owner | OwnerOccupier | Occupier)      => EvidenceChoices.WATER_RATE
+          case (OtherUtilityBill, Owner | OwnerOccupier | Occupier)     => EvidenceChoices.UTILITY_RATE
+          case (Lease, Occupier | Owner | OwnerOccupier | Occupier)     => EvidenceChoices.LEASE
+          case (License, Occupier | Owner | OwnerOccupier | Occupier)   => EvidenceChoices.LICENSE
+          case (_, Owner | OwnerOccupier | Occupier)                    => EvidenceChoices.OTHER
+          case (_, Occupier)                                            => EvidenceChoices.NO_LEASE_OR_LICENSE
```
Comment (joeyymoore) [inline app/controllers/propertyLinking/DeclarationController.scala:85]: this case (_, Occupier) will never happen anymore right? now that youve added the Occupier to the above case statement, so its not needed?

File: app/views/propertyLinking/upload_result.scala.html [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        case (Some(_), Lease) => {
+            messages("uploadEvidence.lease.client.title")
+        }
+        case (Some(_), License) => {
+            messages("uploadEvidence.license.client.title")
+        }
+        case (Some(_), RatesBillType) => {
+            messages("uploadRatesBill.show.client.title")
+        }
+        case (Some(_), _) => {
```
Comment (joeyymoore) [inline app/views/propertyLinking/upload_result.scala.html:65]: is there a client equivalent for this? or is it the same as whats displayed for case (None, _)? Could probably just do a case (_,_) at the end if so

File: app/views/propertyLinking/upload_result.scala.html [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+        case (None, _) => {
+            messages(s"uploadEvidence.$evidenceType.title")
+        }
+    }
+}
+
+@tag(status: FileStatus) = @{
+    status match {
+        case _@FileStatus.UPLOADING => govukTag(Tag(content = Text(messages("fileUploadResult.uploading")), classes = "govuk-tag--yellow", attributes = Map("id" -> "uploadStatus")))
+        case _@FileStatus.READY => govukTag(Tag(content = Text(messages("fileUploadResult.uploaded")), classes = "govuk-tag--green", attributes = Map("id" -> "uploadStatus")))
```
Comment (joeyymoore) [inline app/views/propertyLinking/upload_result.scala.html:86]: is it right that the READY status maps to fileUploadResult.uploaded?

File: conf/logback.xml [hunk 4]
```diff
@@ -12,13 +12,13 @@
...
   <logger name="voa.submission" level="${logger.voa.submission:-DEBUG}"/>
 
-  <logger name="com.google.inject" level="WARN"/>
-  <logger name="reactivemongo" level="WARN"/>
-  <logger name="play" level="WARN"/>
-  <logger name="akka" level="WARN"/>
-  <logger name="uk.gov.hmrc.audit" level="WARN"/>
-  <logger name="auditCounter" level="WARN"/>
-  <logger name="uk.gov.hmrc.play.bootstrap" level="WARN"/>
+<!--  <logger name="com.google.inject" level="WARN"/>-->
```
Comment (joeyymoore) [inline conf/logback.xml:15]: did you mean to leave this and the below commented out?

File: it/controllers/UploadResultControllerISpec.scala [hunk 5]
```diff
@@ -0,0 +1,556 @@
...
+import play.api.test.Helpers._
+import repositories.PropertyLinkingSessionRepository
+import uk.gov.hmrc.http.{HeaderCarrier, SessionId}
+
+import java.time.LocalDate
+import java.util.UUID
+
+class UploadResultControllerISpec extends ISpecBase with HtmlComponentHelpers {
+
+  //  TODO: Add in welsh translations
```
Comment (joeyymoore) [inline it/controllers/UploadResultControllerISpec.scala:23]: still need this TODO?

File: app/views/propertyLinking/upload_result.scala.html (hunk 3 above)
Comment (zachhall1234) [reply app/views/propertyLinking/upload_result.scala.html:86]: Yep, one of the issues with the design of the ticket is that we display uploading/uploaded which aren't accurate to whats happening but yes in this case Ready = Uploaded

File: it/controllers/UploadResultControllerISpec.scala (hunk 5 above)
Comment (zachhall1234) [reply it/controllers/UploadResultControllerISpec.scala:23]: I do not shall revert now cheers

File: app/views/propertyLinking/upload_result.scala.html (hunk 2 above)
Comment (zachhall1234) [reply app/views/propertyLinking/upload_result.scala.html:65]: Good spot, It appears to be the same, if im honest this is a straight copy from the upload.scala however I wonder if we should have content specific for clients for anything that falls in here-> case (Some(_), _)

**Review by joeyymoore — APPROVED** [review]

**Pull request merged**